Fix entrance and exit ghosts not being removed for mazes

mwnciau opened the pull request:
Currently, when placing the entrance or exit of a maze, the ghost of the entrance or exit will not disappear until you actually place it - even if you exit construction. This PR adds a call in the `ride_construction_invalidate_current_track` function to remove the ghost images for mazes when the track is invalidated.

This also solves a separate issue where `RideEntranceExitRemoveAction` would spew a lot of "Track Element not found" warnings in the console.

I couldn't see an existing issue about this. It's not an original bug, but it looks like it's always been in OpenRCT2.

tupaschoal reviewed (commented):
This looks alright, just missing a changelog entry, something like:
Fix: [13074] ...

mwnciau commented:
Added a changelog. I didn't include #13074 because there is no issue for this bug - similar to the loop collision box issue I fixed in 0.3.1 where we decided not to include the PR number.

tupaschoal commented:
> Added a changelog. I didn't include #13074 because there is no issue for this bug - similar to the loop collision box issue I fixed in 0.3.1 where we decided not to include the PR number.

That was an oversight on our part then. We have decide to always have issue or PR # (if first doesn't exist), so one person reading the changelog can quickly go there and read about if, if interested.

tupaschoal commented on distribution/changelog.txt:
```diff
@@ -8,6 +8,7 @@
 - Fix: [#13021] Mowed grass and weeds don't show up in extra zoom levels.
 - Fix: [#13029] Not all Junior Roller Coaster pieces are shown when "Show all track pieces" cheat is enabled.
 - Fix: [#13044] Rides in RCT1 saves all have "0 customers per hour".
+- Fix: Entrance and exit ghosts for mazes not being removed.
```
```suggestion
- Fix: [#13074] Entrance and exit ghosts for mazes not being removed.
```
